src/EventStore/NBB.EventStore/DependencyInjectionExtensions.cs

```diff
@@ -34,5 +38,21 @@ public static IServiceCollection WithNewtownsoftJsonEventStoreSeserializer(this
             return services;
 
         }
-    }
-}
+    }
+
+    internal class ConfigureEventStoreOptions : IConfigureOptions<Abstractions.EventStoreOptions>
+    {
+        private readonly IConfiguration _configuration;
+
+        public ConfigureEventStoreOptions(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public void Configure(Abstractions.EventStoreOptions options)
+        {
+            var connectionString = _configuration.GetSection("EventStore").GetSection("NBB")["ConnectionString"];

```

Comment:
Should also handle TopicSuffix setting
Alternatively use IConfiguration.Get&lt;EventStoreOptions&gt; for automatic binding